typechecker: Missing arguments after defaulted arguments are not reported, leading to C++ compile error

MVS:
```jakt
fn function(a: i32 = 1, b: usize) {}

fn main() {
    function(a: 1)
}
```

Expected this to fail with an error about missing parameter `b`, but it instead fell through:
```
build/defaults.cpp:17:1: error: no matching function for call to 'function'
   17 | function();
      | ^~~~~~~~
build/defaults.cpp:3:6: note: candidate function not viable: requires 2 arguments, but 0 were provided
    3 | void function(i32 const a,size_t const b) {
      |      ^        ~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
Error: Compilation failed
```

--- a/selfhost/typechecker.jakt
+++ b/selfhost/typechecker.jakt
@@ -11174,6 +11174,11 @@ struct Typechecker {
 
                         invalid_considered_argument_uses[consumed_arg].push((i, reason))
                     }
+                } else {
+                    if not param.default_value_expression.has_value() {
+                        .error(format("Missing argument for function parameter '{}'", param.variable.name), span)
+                        continue
+                    }
                 }
             }
 

--- a/tests/typechecker/missing_argument_when_default_argument_involved.jakt
+++ b/tests/typechecker/missing_argument_when_default_argument_involved.jakt
@@ -1,0 +1,10 @@
+/// Expect:
+/// - error: "Missing argument for function parameter 'b'"
+
+fn test(a: i32 = 1, b: usize) {
+    println("Wrong")
+}
+
+fn main() {
+    test(a: 1)
+}

